### DISCUSSION TITLE
POLL: Should I bring the v8 server out earlier (as part of a v7.0.0 release)?

### DISCUSSION THREAD

**Discussion by adrianhall** [Azure/azure-mobile-apps#891] [Polls]
To clarify, I've been working on the v8 server and it has some great features, including System.Text.Json support, isolated JSON encoding (which means that it can coexist with other libraries like ABP), more solid testing and improvements around EF Core support, and GeoJSON support at the server level.  It supports .NET 8 and EF Core 7 out of the box.

However, there are some downsides.  Most notably, there are a bunch of breaking changes associated with the move to System.Text.Json and EF Core 7 that will require everyone that is writing custom repositories to take a look at their code to ensure it is still working.

You can find the code base in the v8 branch.

(Note that the v6 codebase won't go away - I'll branch it and do necessary updates as bugs come in)

This poll will be open until March 2nd.  (I'm going to Ecuador and Galapagos on vacation from Feb 19th - Mar 2nd, so I won't be able to act on this until then anyhow)

**Comment by hansmbakker** [comment]
What would be the overhead of releasing an intermediate version?
And: what would be the rationale of updating to EF Core 7 while it would be possible to skip that and move to EF Core 8, thereby saving time?

If releasing this does not add much of a delay to v8, then I'd be happy to have a v7 - otherwise I'd say I'd rather wait for v8 to have that sooner.

PS: if I misunderstand this and it is not actually an intermediate version, but rather - releasing the server side first and the client later so that we can already upgrade the server - then I would be ok with that.

**Reply by adrianhall** [reply]
Minimal overhead.  There is basically a massive movement of files from the v8 branch (where they now live) to the main branch, and fixing up any problems that the testing with the current client exposes.  It should not be much since the underlying protocol has not changed.

We could definitely update to the latest version of EF Core.

Our client and server versions go lock-step, so it would be upgrading the server code while mostly leaving the client code alone.  Since the protocol has not changed, you can upgrade the server without changing the client.

**Reply by hansmbakker** [reply]
Sounds good!

**Comment by adrianhall** [comment]
Per the votes at this point, I'm going to start working on a v7 branch,
